Bindings/Java/Matlab/Hopper_Device/OptimizeHopper.m

```diff
@@ -25,81 +26,184 @@
+% device stiffness to their maximum. 
 hopper = BuildInteractiveHopperSolution(...
-            'muscleExcitation', excitation, ...
+            'muscle', 'katieLedecky', ...
+            'muscleExcitation', muscleExcitation, ...
             'addPassiveDevice', true, ...
             'passivePatellaWrap', true, ...
-            'passiveParameter', passiveParameter, ...
+            'passiveParameter', 100, ...
             'addActiveDevice', true, ...
             'activePatellaWrap', false, ...
-            'activeParameter', activeParameter, ...
+            'activeParameter', 100, ...
             'deviceControl', deviceControl, ...
             'isActivePropMyo', false, ...
             'printModelInfo', false);
 end
 
-function f = objective(x)
-
-hopper = buildHopperFromSolution(x);
-[peakHeight, ~] = EvaluateHopper(hopper, false, true);
+function [deviceControl, muscleExcitation] = constructControls(x, problem)
+% CONSTRUCTCONTROLS
+%   Construct controls from the optimization optimization variables. Each
+%   variable relates to either:
+%       1) Turning OFF the active device (which releases the loaded spring)  
+%          and turning ON the vastus muscle.
+%       2) Turning ON the active device and turning OFF the vastus muscle
+%          (to reload the spring).
+%   
+%   Each control is described by a 2xN array where the first row contains
+%   time values, the second row contains control values, and N is the 
+%   number of time points:
+%
+%                  control = [t1 t2 ... tN;
+%                             u1 u2 ... uN];
+
+
+% Anonymous functions to conveniently construct 'bang-bang' control signals 
+% for both the active device and the muscle. 
+% 
+% Switch an actuator ON at time 't'. 
+on  =@(t) [t  t+0.01; 
+           0  1.0];
+% Switch an actuator OFF at time 't'.
+off =@(t) [t   t+0.01; 
+           1.0 0];
+       
+% Construct the actuator controls by concatenating a sequence of 'on(t_i)'
+% and 'off(t_i)' anonymous function calls at times t_i = [t1 t2 ... tN].
+% Both problems start with by turning off the muscle and turning on the 
+% active device at t = 0.
+switch problem
+    
+    % The single variable in this problem determines when to simultaneously 
+    % release a loaded passive device (turn OFF the active device) and 
+    % activate the hopper vastus muscle (turn ON the muscle excitation).
+    case 'one_hop'
+        t1 = x;
+        deviceControl    = [on(0)  off(t1)];
+        muscleExcitation = [off(0) on(t1)];
+        
+    % To coordinate the sequence of hops in this problem, the spring must
+    % be loaded and released (with the activated muscle) three times. The
+    % first variable, x(1), is the wait time between when the spring is 
+    % loaded and released for each hop. The x(1) value can be used for all 
+    % three hops since we also optimize the wait time between each hop with 
+    % the other two variables, x(2) and x(3). The active device is turned 
```

Comment:
I had to read this a few times to understand. How about "The x(1) value is the duration for which the spring is loaded before each hop. The variables x(2) and x(3) are  the wait times to start loading the spring again for the 2nd and 3rd hops, respectively."

Reply:
I like this rewording. Mine definitely wasn't clear on second read.